vendor/bat-native-ads/src/bat/ads/internal/rpill/rpill_helper.cc

```diff
@@ -0,0 +1,81 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ads/internal/rpill/rpill_helper.h"
+
+#include <string>
+#include <vector>
+
+#include "base/strings/string_util.h"
+#include "base/system/sys_info.h"
+#include "bat/ads/internal/logging.h"
+#include "bat/ads/internal/sys_info_helper.h"
+
+namespace ads {
+
+namespace {
+
+std::string GetHardware() {
+  const base::SysInfo::HardwareInfo hardware =
```

Comment:
you should have a DCHECK before calling `SysInfoHelper::GetInstance()->GetHardware()` to ensure that it's ready